x/mongo/driver/topology/rtt_monitor.go

```diff
@@ -190,6 +195,31 @@ func min(samples []time.Duration, minSamples int) time.Duration {
 	return min
 }
 
+// percentile90 returns the 90th percentile value of the slice of duration samples. Zero values
+// are not considered samples and are ignored. If no samples or fewer than minSamples are found
+// in the slice, percentile90 returns 0.
+func percentile90(samples []time.Duration, minSamples int) time.Duration {
+	// Convert Durations to float64s.
+	var count int
+	floatSamples := make([]float64, 0)
+	for _, sample := range samples {
+		if sample > 0 {
+			count++
+		}
+
+		floatSamples = append(floatSamples, float64(sample))
+	}
+	if count < minSamples {
```

Comment:
I think we want to return `0` when the number of _non-zero_ samples is less than `minSamples`. We have to maintain a `count` for that, which is distinct from the length of the slice.

Reply:
Ah sorry, this was really unclear on my part.  But, I think you can do this in the loop:

```go
if sample > 0 {
	floatSamples = append(floatSamples, float64(sample))
}
```

The reasoning is that one of two things is done to determine the percentile under the hood:

1. The [nearest rank](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method) is a natural number, [in which case they take the largest number in the sample](https://github.com/montanaflynn/stats/blob/v0.6.6/percentile.go#L35).
2. The nearest rank is not a natural number, [in which case they take the mean of the sample](https://github.com/montanaflynn/stats/blob/v0.6.6/percentile.go#L43).

So there can be any number of zero values in the sample, and the result of the percentile will be the same with or without them.  For example, both `p1` and `p2` will both result in 5:
```
p1, _ := stats.Percentile([]float64{0, 1, 9, 0, 0, 0}, 90)
p2, _ := stats.Percentile([]float64{1, 9}, 90)
fmt.Println(p1, p2) // 5 5
```

I'm suggesting we just don't include the zero values at all, since they don't matter in either calculation and excluding them makes the code more straightforward IMO.

Reply:
Ah, after discussing offline I see what you're saying. Thanks for the extra explanation! I now only append samples that are greater than `0` and check that `len(floatSamples) > minSamples`.